lib/dialects/postgres/query-generator.js

```diff
@@ -347,8 +347,10 @@ class PostgresQueryGenerator extends AbstractQueryGenerator {
     const insert = this.insertQuery(tableName, insertValues, model.rawAttributes, upsertOptions);
     const update = this.updateQuery(tableName, updateValues, where, upsertOptions, model.rawAttributes);
 
-    insert.query = insert.query.replace('RETURNING *', `RETURNING ${primaryField} INTO primary_key`);
-    update.query = update.query.replace('RETURNING *', `RETURNING ${primaryField} INTO primary_key`);
+    if (options.returning) {
+      insert.query = insert.query.replace(/RETURNING \*(?![\s\S]*RETURNING \*)/, `RETURNING ${primaryField} INTO primary_key`);
```

Comment:
Nit: Regex should be moved to a constant.

Reply:
Done.